ddtrace/_trace/telemetry.py

```diff
@@ -0,0 +1,29 @@
+from typing import Optional
+from typing import Tuple
+
+from ddtrace.internal.telemetry import telemetry_writer
+
+
+def record_span_pointer_calculation(context: str, span_pointer_count: int) -> None:
+    telemetry_writer.add_count_metric(
+        namespace="tracer", name="span_pointer_calculation", value=1, tags=(("context", context),)
+    )
+
+    telemetry_writer.add_count_metric(
```

Comment:
I wonder if the count would better fit as a tag on the span_pointer_calculation metric so we only need to send one metric? It obviously adds cardinality... but do we think the count will actually be that high ever? Like if we don't expect more than 20 for the count, I think it's totally fine to add that as tag instead of using an additional metric. Although maybe that goes against some metric rule I don't know about.

Reply:
the count can be arbitrarily high, though in apis we currently support i think the realistic limit might be 10

Reply:
though we can do something like [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16-20, 21-50, 51-100, 101+]? (i.e. bucket the large counts)

Reply:
(i've done something like this)